vpc/globalgc/globalgc.go

```diff
@@ -4,303 +4,175 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/Netflix/titus-executor/vpc"
 	"github.com/Netflix/titus-executor/vpc/context"
-	"github.com/Netflix/titus-executor/vpc/setup"
+	"github.com/Netflix/titus-executor/vpc/ec2util"
 	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"gopkg.in/urfave/cli.v1"
 )
 
-const (
-	markTag = "titus-gc-mark-time"
-)
-
-var GlobalGC = cli.Command{ // nolint: golint
-	Name:   "globalgc",
-	Usage:  "Garbage collect detached ENIs",
-	Action: context.WrapFunc(globalGc),
-	Flags: []cli.Flag{
-		cli.DurationFlag{
-			Name:  "timeout",
-			Usage: "Maximum amount of time allowed running GC",
-			Value: time.Minute * 5,
-		},
-		cli.DurationFlag{
-			Name:  "detach-time",
-			Usage: "How long an ENI has to be detached before we will clean it up",
-			Value: time.Minute * 30,
-		},
-		cli.StringFlag{
-			Name:   "vpc-id",
-			Usage:  "Optionally specify a VPC, to speed up filtering requests",
-			EnvVar: "EC2_VPC_ID",
-			Value:  "",
-		},
-	},
+type globalGcConfig struct {
+	timeout    time.Duration
+	detachTime time.Duration
+	vpcID      string
+	workers    int
 }
 
-func globalGc(parentCtx *context.VPCContext) error {
-	timeout := parentCtx.CLIContext.Duration("timeout")
-	ctx, cancel := parentCtx.WithTimeout(timeout)
-	defer cancel()
-
-	minDetachTime := parentCtx.CLIContext.Duration("detach-time")
-	vpcID := parentCtx.CLIContext.String("vpc-id")
-
-	if err := doGlobalGc(ctx, minDetachTime, vpcID); err != nil {
-		return cli.NewMultiError(cli.NewExitError("Unable to run GC", 1), err)
+// GlobalGC is the configuration for the command that deletes ENI which have been unattached too long
+func GlobalGC() cli.Command {
+	cfg := &globalGcConfig{}
+	globalGc := func(parentCtx *context.VPCContext) error {
+		if err := doGlobalGC(parentCtx, cfg); err != nil {
+			return cli.NewMultiError(cli.NewExitError("Unable to run global gc", 1), err)
+		}
+		return nil
+	}
+	return cli.Command{ // nolint: golint
+		Name:   "globalgc",
+		Usage:  "Garbage collect detached ENIs",
+		Action: context.WrapFunc(globalGc),
+		Flags: []cli.Flag{
+			cli.DurationFlag{
+				Name:        "timeout",
+				Usage:       "Maximum amount of time allowed running GC",
+				Value:       time.Minute * 5,
+				Destination: &cfg.timeout,
+			},
+			cli.DurationFlag{
+				Name:        "detach-time",
+				Usage:       "How long an ENI has to be detached before we will clean it up",
+				Value:       time.Minute * 30,
```

Comment:
now that this represents "how long since disconnected ENIs were created", should this be bumped to a few days? It seems we will be potentially cleaning up ENIs that have been in use for more than 30min much earlier now. Or is it also the intent?

Reply:
If we try to delete an ENI and it's currently attached, the call fails.